pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -211,7 +211,19 @@ spec:
         istio.io/test-vm-version: {{ $subset.Version }}
     spec:
       # Disable kube-dns, to mirror VM
-      dnsPolicy: Default
+      # we set policy to none and explicitly provide a set of invalid values
+      # for nameservers, search namespaces, etc. ndots is set to 1 so that
+      # the application will first try to resolve the hostname (a, a.ns, etc.) as is
+      # before attempting to add the search namespaces.
+      dnsPolicy: None
+      dnsConfig:
+        nameservers:
+        - "8.8.8.8"
```

Comment:
to make sure that vm can never resolve dns on its own without envoy

Reply:
Why do we need all of this? Then `dnsPolicy: Default` already ensures we do not use kube-dns. I am worried we are now testing something different than what will run on a real VM?

Reply:
Its not just about using kube-dns. We need to make sure that there is a valid dns resolver. Envoy segfaults otherwise. With the default policy, the resolver is set to 169.254.169.254 inside kind, which is not what a real VM has. So using the google resolver 8.8.8.8 is atleast a good approximation of what a VM will have, and we can also ensure that it wont resolve k8s services.

Reply:
good point, it's good to be consistent anyways so all envs have the same DNS config